packages/core/admin/admin/src/pages/Roles/ListPage/index.js

```diff
@@ -1,147 +1,186 @@
-import { Button } from '@buffetjs/core';
-import { Duplicate, Pencil, Plus } from '@buffetjs/icons';
+import { useQuery, useTracking } from '@strapi/helper-plugin';
+import { AddIcon, DeleteIcon, EditIcon, Duplicate } from '@strapi/icons';
+import {
+  Button,
+  ContentLayout,
+  HeaderLayout,
+  Table,
+  TableLabel,
+  Tbody,
+  TFooter,
+  Th,
+  Thead,
+  Tr,
+  VisuallyHidden,
+} from '@strapi/parts';
 import matchSorter from 'match-sorter';
+import React, { useCallback, useState } from 'react';
 import { useIntl } from 'react-intl';
-import { useHistory } from 'react-router-dom';
-import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
-import { ListButton, useTracking, useQuery, useRBAC } from '@strapi/helper-plugin';
-import adminPermissions from '../../../permissions';
+import { useHistory } from 'react-router';
+import { EmptyRole, RoleRow } from '../../../components/Roles';
 import PageTitle from '../../../components/SettingsPageTitle';
-import { EmptyRole, RoleListWrapper, RoleRow } from '../../../components/Roles';
-import { useRolesList, useSettingsHeaderSearchContext } from '../../../hooks';
 import UpgradePlanModal from '../../../components/UpgradePlanModal';
-import BaselineAlignment from './BaselineAlignment';
+import { useRolesList } from '../../../hooks';
 
-const RoleListPage = () => {
-  const { formatMessage } = useIntl();
-  const { push } = useHistory();
-  const [isOpen, setIsOpen] = useState(false);
-  const { trackUsage } = useTracking();
+const useResults = () => {
   const { roles, isLoading } = useRolesList();
-  const { toggleHeaderSearch } = useSettingsHeaderSearchContext();
-  const {
-    allowedActions: { canUpdate },
-  } = useRBAC(adminPermissions.settings.roles);
+
   const query = useQuery();
   const _q = decodeURIComponent(query.get('_q') || '');
   const results = matchSorter(roles, _q, { keys: ['name', 'description'] });
 
-  useEffect(() => {
-    toggleHeaderSearch({ id: 'Settings.permissions.menu.link.roles.label' });
+  return { isLoading, results };
+};
 
-    return () => {
-      toggleHeaderSearch();
-    };
-    // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, []);
+const useFuncs = () => {
+  const { formatMessage } = useIntl();
+  const [isModalOpen, setIsModalOpen] = useState(false);
+  const { trackUsage } = useTracking();
+  const { push } = useHistory();
 
   const handleGoTo = useCallback(
     id => {
       push(`/settings/roles/${id}`);
     },
-    // eslint-disable-next-line react-hooks/exhaustive-deps
-    []
+    [push]
   );
 
-  const handleToggle = useCallback(e => {
-    e.preventDefault();
-    e.stopPropagation();
-    setIsOpen(prev => !prev);
+  const handleToggle = useCallback(() => {
+    setIsModalOpen(prev => !prev);
   }, []);
 
-  const handleToggleModalForCreatingRole = useCallback(e => {
-    e.preventDefault();
-    e.stopPropagation();
+  const handleToggleModalForCreatingRole = useCallback(() => {
```

Comment:
This func is used in a hook. Do you prefer to add // eslint-disable-next-line react-hooks/exhaustive-deps

Reply:
I meant:

```jsx
const handleToggleModalForCreatingRole = () => {
  trackUsage('didShowRBACUpgradeModal');
   setIsModalOpen(true);
}
```

instead of wrapping it in a hook :D